skymap_scanner/server/pixels.py

```diff
@@ -182,13 +182,13 @@ def choose_pixels_to_reconstruct(
 
     Some of the pixel returned may have previously been generated.
     """
-    # special case if we have MC truth
-    if mc_ra_dec:
-        LOGGER.debug("Getting pixels around MC truth...")
-        # scan only at max_nside around the true minimum
-        return choose_pixels_to_reconstruct_around_MCtruth(
+    # special case if we have a given coordinate, select pixels only at first nside
+    if coord_ra_dec and len(nside_progression) == 1 :
```

Comment:
What happens if `coord_ra_dec` is set but `len(nside_progression) !=1`? Maybe this should be an `assert` instead so that it fails if the user passes multiple sides?

Reply:
If that's the assumption, then let's (also) put that validation check at the very top of the pipeline where the CL args are retrieved

Reply:
The function `choose_pixels_to_reconstruct_around_coord` must run only when the pixels for the first `nside` are chosen, and not when pixels are selected for refinements. When the scanner chooses the pixels for higher nsides, then `coord_ra_dec` is set but  `len(nside_progression) !=1 (here the `nside_progression` is what in `start_scan.py` is the `nside_subprogression`, and at the first iteration is just a `dict` with `len(nside_subprogression) = 1`), and this is necessary as we discussed here: https://github.com/icecube/skymap_scanner/pull/265#discussion_r1553791298.

Reply:
> If that's the assumption, then let's (also) put that validation check at the very top of the pipeline where the CL args are retrieved

`start_scan.py` calls `choose_pixels_to_reconstruct()` sending the `nside_subprogression`, which just tells until which `nside` the pixels must be selected. The `nside_progression` that is sent as an input to `start_scan` is the complete progression and is not what is checked here, so I would avoid a validation check at the beginning of the pipeline, as the `nside_progression` that is sent as an arg is not the `nside_progression` that is checked here.

Reply:
I believe I understand--in other words, `len(nside_subprogression)` may be > 1 for scans that define `coord_ra_dec` (aka `splinempe_pointed`). Is that correct?

Reply:
Yes, at all the pixel refinements `len(nside_subprogression)` will be `>1` and the pixels won't be chosen by `choose_pixels_to_reconstruct_around_coord()`